PowerShell/BuildingBlocks/CustomerAndResources/AssignSubscriptions.ps1

```diff
@@ -0,0 +1,22 @@
+﻿# Initialize the context
+.\Init.ps1 -Customer
+
+# Get the end users under that customer (this example retrieves the top 100)
+$customerEndUsers =  Get-BT_CustomerEndUser -Ticket $mspc.CustomerTicket -OrganizationId $mspc.Customer.OrganizationId -IsDeleted $False -PageSize 100
+if ( $customerEndUsers ) {
+    # Check for existing subscriptions
+    $existingSubscriptions = Get-BT_Subscription -Ticket $mspc.WorkgroupTicket -SubscriptionEntityReferenceType CustomerEndUser -EntityReferenceId $customerEndUsers.Id -IsDeleted $False
+    
+    # Filter out end users who already have a subscription
+    $customerEndUsersToSubscribe = $customerEndUsers | Where {
+        !( $existingSubscriptions.EntityReferenceId -contains $_.Id ) 
```

Comment:
May be more clear to use '-notcontains' operator and remove the negation.

Reply:
fixed